File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/javascript/cwe-77/code_before3.js
// From cwe-snippets, snippets_2/non-compliant/JavaScript/0009.js

var cp = require('child_process');
var http = require('http');
var url = require('url');
function listener(request, response){
  var btype = url.parse(request.url, true)['query']['backuptype'];
  if (btype !== undefined){
    cmd = "c:\\util\\rmanDB.bat" + btype;
    cp.exec(cmd, function(error, stdout, stderr){
    });
  }
}
http.createServer(listener).listen(8080);

File: experiments_dataset_vul_detect_cwe_classification/dataset/vul_dataset/python/cwe-20/code_before3.py
# From cwe-snippets, snippets_2/non-compliant/Python/0045.py

def store(request):
    id = request.GET['id']
    result = get_page_from_somewhere()
    response = HttpResponse(result)
    cache_time = 1800
    cache.set("req-" % id, response, cache_time)
    return response